dex-it/src/test/scala/com/wavesplatform/it/sync/OrderExecutedDataInProofsTestSuite.scala

```diff
@@ -0,0 +1,173 @@
+import com.wavesplatform.dex.it.test.Scripts
+import com.wavesplatform.it.MatcherSuiteBase
+import com.wavesplatform.transactions.exchange
+
+import scala.jdk.CollectionConverters._
+
+final class OrderExecutedDataInProofsTestSuite extends MatcherSuiteBase {
+
+  private val orderVersion = 3.toByte
+
+  "OrderExecutedDataInProofs" - {
+
+    "before offset don't add extra data for any account" in {
+      val aliceOrder = mkOrder(alice, wavesUsdnPair, OrderType.SELL, 1.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(aliceOrder)
+      val carolOrder = mkOrder(carol, wavesUsdnPair, OrderType.BUY, 1.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(carolOrder)
+
+      dex1.api.waitForOrderStatus(carolOrder, HttpOrderStatus.Status.Filled)
+
+      val txs = dex1.api.getTransactionsByOrderId(carolOrder)
+      txs.size shouldBe 1
+      val orders = txs.head.orders()
+      orders.size() shouldBe 2
+      orders.asScala.foreach(_.proofs().size() shouldBe 1)
+    }
+
+    "add extra data for old order if offset is right" in {
+      val bobOrder = mkOrder(bob, wavesUsdnPair, OrderType.BUY, 5.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(bobOrder)
+
+      val aliceOrder1 = mkOrder(alice, wavesUsdnPair, OrderType.SELL, 4.3.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(aliceOrder1)
+
+      val txs1 = dex1.api.getTransactionsByOrderId(aliceOrder1)
+      txs1.size shouldBe 1
+      val orders1 = txs1.head.orders()
+      orders1.size() shouldBe 2
+      orders1.asScala.foreach(_.proofs().size() shouldBe 1)
+
+      broadcastAndAwait(
+        mkSetAccountMayBeScript(alice, Some(script), fee = setScriptFee + smartFee),
+        mkSetAccountMayBeScript(bob, Some(script), fee = setScriptFee + smartFee)
+      )
+
+      val aliceOrder2 = mkOrder(alice, wavesUsdnPair, OrderType.SELL, 0.7.waves, 1.3.usdn, version = orderVersion)
+      dex1.api.place(aliceOrder2)
+
+      val txs2 = dex1.api.getTransactionsByOrderId(aliceOrder2)
+      txs2.size shouldBe 1
+      val orders2 = txs2.head.orders()
+      orders2.size() shouldBe 2
+      orders2.asScala.foreach(_.proofs().size() shouldBe 3)
+    }
+
+    "after offset add extra data only for account from config" in {
+      val aliceOrder = mkOrder(alice, wavesUsdnPair, OrderType.SELL, 1.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(aliceOrder)
+      val carolOrder = mkOrder(carol, wavesUsdnPair, OrderType.BUY, 1.waves, 2.usdn, version = orderVersion)
+      dex1.api.place(carolOrder)
+
+      dex1.api.waitForOrderStatus(carolOrder, HttpOrderStatus.Status.Filled)
+
+      val txs = dex1.api.getTransactionsByOrderId(carolOrder)
+      txs.size shouldBe 1
+      val orders = txs.head.orders()
+      orders.size() shouldBe 2
+      orders.asScala.foreach { order =>
+        if (order.`type`() == exchange.OrderType.BUY) order.proofs().size() shouldBe 1
+        else order.proofs().size() shouldBe 3
```

Comment:
i think we can do more complex checks in order to validate that exec price/amount was passed correctly